cmd/kanikoExecute.go

```diff
@@ -134,117 +135,207 @@ func runKanikoExecute(config *kanikoExecuteOptions, telemetryData *telemetry.Cus
 	}
 	commonPipelineEnvironment.custom.buildSettingsInfo = buildSettingsInfo
 
-	if !piperutils.ContainsString(config.BuildOptions, "--destination") {
-		dest := []string{"--no-push"}
-		if len(config.ContainerRegistryURL) > 0 && len(config.ContainerImageName) > 0 && len(config.ContainerImageTag) > 0 {
-			containerRegistry, err := docker.ContainerRegistryFromURL(config.ContainerRegistryURL)
-			if err != nil {
-				log.SetErrorCategory(log.ErrorConfiguration)
-				return errors.Wrapf(err, "failed to read registry url %v", config.ContainerRegistryURL)
+	switch {
+	case config.ContainerMultiImageBuild:
```

Comment:
This case expects that the containerRegistryURL, containerImageName, and containerImageTag are not empty. Probably it makes sense to add checking.